code/unclassified/src/segregate_even_odd/segregate_even_odd.cpp

```diff
@@ -0,0 +1,47 @@
+// Segregate all even numbers to left and odd numbers to right of the array
+#include <algorithm>
+#include <iostream>
+#include <stdlib.h>
+#include <vector>
+
+void even_odd(std::vector<int> a, int n) {
+    int i = 0;
+    int j = n - 1;
+    while (i < j) {
+        while (a[i] % 2 == 0 && i < j)
+            i++;
+        while (a[j] % 2 != 0 && i < j)
+            j--;
+        if (i < j) {
+            std::swap(a[i], a[j]);
+            i++;
+            j--;
+        }
+    }
+
+    std::cout << "After segregation : \n";
+    for (int i = 0; i < n; ++i)
+        std::cout << a[i] << " ";
+}
+
+int main() {
+    int n;
+    std::cout << "Enter the number of elements in array :  ";
+    std::cin >> n;
+    std::vector<int> a(n);
+    std::cout << "Enter the values of the elements :  ";
+    for (int i = 0; i < n; ++i) {
+        int val;
+        std::cin >> val;
+        a.push_back(val);
```

Comment:
This line needs to be changed still.

Reply:
I cannot write a[ i ] = val; Since a is a vector

Reply:
I will get an error if i change this line

Reply:
`std::vector` supports the `[]` operator: https://en.cppreference.com/w/cpp/container/vector/operator_at. Otherwise this code adds past the end of the array and you end up with a vector twice the expected size.

Reply:
Also anywhere you need the size of the array, use `.size()` instead of passing `n` around.

Reply:
Got it. Made the changes requested